storage/configuration.go

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2025 Eric Faurot <eric@faurot.net>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+package storage
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/vmihailenco/msgpack/v5"
+
+	"github.com/PlakarKorp/plakar/compression"
+)
+
+func newConfigurationFromBytes(buffer []byte, format string) (*Configuration, error) {
+
+	if format == "auto" {
+		// autodetect
+		formats := []string{"msgpack", "json", "json+gz"}
+		for _, format := range formats {
+			config, err := newConfigurationFromBytes(buffer, format)
+			if err == nil {
+				return config, nil
+			}
+		}
+		return nil, fmt.Errorf("Could not guess configuration format")
+	}
+
+	var err error
+	var unmarshal func(data []byte, v interface{}) error
+	gzip := false
+
+	switch format {
+	case "msgpack":
+		gzip = true
+		unmarshal = msgpack.Unmarshal
+	case "json":
+		unmarshal = json.Unmarshal
+	case "json+gz":
```

Comment:
is this used ? if you json+gz, i'm not sure why you would not use msgpack instead ?